tests/args.py

```diff
@@ -164,6 +169,42 @@ def test_commands_together(self):
         self.assertEqual(args.commands, [["test", "args"], ["restart", "all"]])
 
 
+class TestProfileArguments(unittest.TestCase):
+
```

Comment:
I will clarify that I have tested this by hand in my container test environment.  I can demonstrate for anyone if you want to confirm specific use cases.  I had to fix some details after stress testing this and am more than welcome if there are any worries about things I may have missed.